ydb/src/load_balancer.rs

```diff
@@ -312,4 +706,473 @@ mod test {
         assert!(*map.get(two.to_string().as_str()).unwrap() > 30);
         Ok(())
     }
+
+    #[test]
+    fn split_by_location() -> YdbResult<()> {
+        let nodes = vec![
+            NodeInfo::new(Uri::from_str("http://one:213")?, "A".to_string()),
+            NodeInfo::new(Uri::from_str("http://two:213")?, "A".to_string()),
+            NodeInfo::new(Uri::from_str("http://three:213")?, "B".to_string()),
+            NodeInfo::new(Uri::from_str("http://four:213")?, "B".to_string()),
+            NodeInfo::new(Uri::from_str("http://five:213")?, "C".to_string()),
+        ];
+        let splitted = NearestDCBalancer::split_endpoints_by_location(&nodes);
+        assert_eq!(splitted.keys().len(), 3);
```

Comment:
What about compare with exact map instead of check lens?

What about the code push good numbers of items, but all items is bad uris, or same uri?